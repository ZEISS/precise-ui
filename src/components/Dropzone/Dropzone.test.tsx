import * as React from 'react';
import * as enzyme from 'enzyme';
import { Dropzone } from './';

describe('<Dropzone />', () => {
  it('should render <Dropzone> that is implicitly not controlled', () => {
    const wrapper = enzyme.mount(enzyme.shallow(<Dropzone />).get(0));
    expect(wrapper.state('controlled')).toBe(false);
  });

  it('should render <Dropzone> that is explicitly not controlled', () => {
    const wrapper = enzyme.mount(enzyme.shallow(<Dropzone defaultValue={[]} />).get(0));
    expect(wrapper.state('controlled')).toBe(false);
  });

  it('should render a <Dropzone> that is controlled', () => {
    const wrapper = enzyme.mount(enzyme.shallow(<Dropzone value={[]} />).get(0));
    expect(wrapper.state('controlled')).toBe(true);
  });

  it('should render a <Dropzone> with an error', () => {
    const wrapper = enzyme.shallow(<Dropzone error="You need to select a file" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a disabled <Dropzone>', () => {
    const wrapper = enzyme.shallow(<Dropzone disabled />);
    expect(wrapper).toMatchSnapshot();
  });
});
