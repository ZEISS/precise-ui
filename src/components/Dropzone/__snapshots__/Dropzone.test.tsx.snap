// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropzone /> should render a <Dropzone> with an error 1`] = `
<DropzoneInt
  error="You need to select a file"
/>
`;

exports[`<Dropzone /> should render a disabled <Dropzone> 1`] = `
<DropzoneInt
  disabled={true}
/>
`;
